dice_ml/explainer_interfaces/dice_genetic.py

```diff
@@ -305,151 +288,78 @@ def predict_fn_scores(self, input_instance):
     def predict_fn(self, input_instance):
         input_instance = self.label_decode(input_instance)
         # TODO this line needs to change---we should not call model.model directly here. That functionality should be in the model class
-        output = self.model.model.predict(input_instance)[0]
+        output = self.model.model.predict(input_instance)
         return output
 
     def compute_yloss(self, cfs, desired_range, desired_class):
         """Computes the first part (y-loss) of the loss function."""
         yloss = 0.0
         if self.model.model_type == 'classifier':
+            predicted_value = np.array(self.predict_fn_scores(cfs))
             if self.yloss_type == 'hinge_loss':
-                for i in range(self.total_CFs):
-                    predicted_values = self.predict_fn_scores(cfs[i])[0]
-
-                    maxvalue = -np.inf
-                    for c in range(self.num_output_nodes):
-                        if c != desired_class:
-                            maxvalue = max(maxvalue, predicted_values[c])
-                    temp_loss = max(0, maxvalue - predicted_values[int(desired_class)])
-                    yloss += temp_loss
-            return yloss/self.total_CFs
+                maxvalue = np.full((len(predicted_value)), -np.inf)
+                for c in range(self.num_output_nodes):
+                    if c != desired_class:
+                        maxvalue = np.maximum(maxvalue, predicted_value[:, c])
+                yloss = np.maximum(0, maxvalue - predicted_value[:, int(desired_class)])
+            return yloss
 
         elif self.model.model_type == 'regressor':
+            predicted_value = self.predict_fn(cfs)
             if self.yloss_type == 'hinge_loss':
-                for i in range(self.total_CFs):
-                    predicted_value = self.predict_fn(cfs[i])
-                    if desired_range[0] <= predicted_value <= desired_range[1]:
-                        temp_loss = 0
-                    else:
-                        temp_loss = min(abs(predicted_value - desired_range[0]), abs(predicted_value - desired_range[1]))
-                    yloss += temp_loss
-            return yloss / self.total_CFs
+                yloss = np.zeros(len(predicted_value))
+                for i in range(len(predicted_value)):
+                    if not desired_range[0] <= predicted_value[i] <= desired_range[1]:
+                        yloss[i] = min(abs(predicted_value[i] - desired_range[0]), abs(predicted_value[i] - desired_range[1]))
+            return yloss
 
-    def compute_dist(self, x_hat, x1):
+    def compute_proximity_loss(self, x_hat):
```

Comment:
what about cont versus categorical features? does the same loss work?
if so, good to write a comment in line 316 on how it works for both kinds of features.